packages/talker_grpc_logger/lib/src/talker_grpc_logger_base.dart

```diff
@@ -0,0 +1,190 @@
+        method: method,
+        request: request,
+        options: options,
+        obfuscateToken: obfuscateToken));
+
+    DateTime startTime = DateTime.now();
+    final response = invoker(method, request, options);
+
+    response.then((r) {
+      Duration elapsedTime = DateTime.now().difference(startTime);
+      _talker.logTyped(GrpcResponseLog(method.path,
+          method: method, response: r, durationMs: elapsedTime.inMilliseconds));
+    }).catchError((e) {
+      Duration elapsedTime = DateTime.now().difference(startTime);
+      _talker.logTyped(GrpcErrorLog(method.path,
+          method: method,
+          request: request,
+          options: options,
+          grpcError: e,
+          durationMs: elapsedTime.inMilliseconds,
+          obfuscateToken: obfuscateToken));
+    });
+    return response;
+  }
+
+  @override
+  ResponseStream<R> interceptStreaming<Q, R>(
+      ClientMethod<Q, R> method,
+      Stream<Q> requests,
+      CallOptions options,
+      ClientStreamingInvoker<Q, R> invoker) {
+    print('interceptStreaming');
+
+    return invoker(method, requests, options);
+  }
+}
+
+class GrpcRequestLog<Q, R> extends TalkerLog {
+  GrpcRequestLog(
+    String title, {
+    required this.method,
+    required this.request,
+    required this.options,
+    this.obfuscateToken = true,
+  }) : super(title);
+
+  final ClientMethod<Q, R> method;
+  final Q request;
+  final CallOptions options;
+  final bool obfuscateToken;
+
+  @override
+  AnsiPen get pen => (AnsiPen()..xterm(219));
+
+  @override
+  String get title => 'grpc-request';
+
+  @override
+  String generateTextMessage() {
+    var time = TalkerDateTimeFormatter(DateTime.now()).timeAndSeconds;
+    var msg = '[$title] | $time | [${method.path}]';
+
+    msg += '\nRequest: ${request.toString().replaceAll("\n", " ")}';
+
+    // Add the headers to the log message, but obfuscate the token if
+    // necessary.
+    final Map<String, String> headers = {};
+    options.metadata.forEach((key, value) {
+      if (obfuscateToken && key.toLowerCase() == 'authorization') {
+        headers[key] = 'Bearer [obfuscated]';
+      } else {
+        headers[key] = value;
+      }
+    });
+
+    try {
+      if (headers.isNotEmpty) {
+        final prettyHeaders = encoder.convert(headers);
+        msg += '\nHeaders: $prettyHeaders';
+      }
+    } catch (_) {
+      // TODO: add handling can`t convert
+    }
+    return msg;
+  }
+}
+
+class GrpcErrorLog<Q, R> extends TalkerLog {
+  GrpcErrorLog(
+    String title, {
+    required this.method,
+    required this.request,
+    required this.options,
+    required this.grpcError,
+    required this.durationMs,
+    this.obfuscateToken = true,
+  }) : super(title);
+
+  final ClientMethod<Q, R> method;
+  final Q request;
+  final CallOptions options;
+  final GrpcError grpcError;
+  final int durationMs;
+  final bool obfuscateToken;
+
+  @override
+  AnsiPen get pen => (AnsiPen()..xterm(219));
```

Comment:
Please replace error log color with red like in talker_dio_logger

Reply:
Done